session/middleware_metrics.go

```diff
@@ -0,0 +1,56 @@
+package session
+
+import (
+	"context"
+	"time"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kit/metric"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// SessionMetrics is a metrics middleware system for the session service
+type SessionMetrics struct {
+	// RED metrics
+	rec *metric.REDClient
+
+	sessionSvc influxdb.SessionService
+}
+
+var _ influxdb.SessionService = (*SessionMetrics)(nil)
+
+// NewSessionMetrics creates a new session metrics middleware
+func NewSessionMetrics(reg prometheus.Registerer, s influxdb.SessionService) *SessionMetrics {
+	return &SessionMetrics{
+		rec:        metric.New(reg, "session"),
+		sessionSvc: s,
+	}
+}
+
+// FindSession calls the underlieing session service and tracks RED metrics for the call
+func (m *SessionMetrics) FindSession(ctx context.Context, key string) (session *influxdb.Session, err error) {
+	rec := m.rec.Record("Find_session")
```

Comment:
Curiosity comment: Is this metric name in line with other services. Just interested in the name, it is kind of mix of camel and snake case.

Reply:
your right here snake case is the way we have been doing it. Making a fix now.